src/sentry/grouping/enhancer.py

```diff
@@ -245,12 +302,35 @@ def apply_modifications_to_frame(self, frames, platform):
                     action.apply_modifications_to_frame(frames, idx)
 
     def update_frame_components_contributions(self, components, frames, platform):
+        stack_state = StackState()
+
+        # Apply direct frame actions and update the stack state alongside
         for rule in self.iter_rules():
             for idx, (component, frame) in enumerate(izip(components, frames)):
                 actions = rule.get_matching_frame_actions(frame, platform)
                 for action in actions or ():
                     action.update_frame_components_contributions(
                         components, idx, rule=rule)
+                    action.modify_stack_state(stack_state, rule)
+
+        # Use the stack state to update frame contributions again
+        max_frames = stack_state.get('max-frames')
+        if max_frames is not None:
+            ignored = 0
+            for component in reversed(components):
+                if not component.contributes:
+                    continue
+                ignored += 1
+                if ignored <= max_frames:
+                    continue
+                hint = 'ignored because only %d %s considered' % (
```

Comment:
_nit_: Would it be possible to execute a rule containing `FlagAction('group', True, None)` rather than hard-coding this here? Seems like this could be "composed" a little nicer.

Reply:
Yes, but then the hint would be different. This one I thought reads nicer.